src/user_override.cc

```diff
@@ -10,26 +10,62 @@
 #include "user_override.hpp"
 
 #include "logger.hpp"
-#include "unistd.h"
 
-#include <errno.h>
+#include <cerrno>
+#include <grp.h>
 #include <pwd.h>
 #include <string.h>
 #include <sys/prctl.h>
+#include <sys/types.h>
 #include <unistd.h>
 
-static char const *const s_user_nobody = "nobody";
-static const uid_t s_root_user = 0;
+namespace {
+constexpr const char *const s_user_nobody = "nobody";
+constexpr uid_t s_root_user = 0;
 
-void init_uidinfo(UIDInfo *user_override) {
-  user_override->override = false;
-  user_override->previous_user = getuid();
+struct DumpableRestorer {
```

Comment:
outcome is nice for returning a failure or a value, here it's pure RAII without error handling (failures are not relevant, it's a best effort), therefore there is no much benefit to using benefit.
It would had made sense to use outcome for `FileHolder` to return an error instead of an empty `unique_ptr`, but I prefer to introduce outcome in a later PR.